bundles/core/src/main/java/com/adobe/cq/commerce/core/cacheinvalidation/spi/DispatcherCacheInvalidationContext.java

```diff
@@ -0,0 +1,86 @@
+/*******************************************************************************
+ *
+ *    Copyright 2025 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+
+package com.adobe.cq.commerce.core.cacheinvalidation.spi;
+
+import java.util.Map;
+
+import org.apache.sling.api.resource.ResourceResolver;
+
+import com.adobe.cq.commerce.graphql.client.GraphqlClient;
+import com.day.cq.wcm.api.Page;
+
+/**
+ * Context object containing all parameters needed for dispatcher cache invalidation.
+ */
+public class DispatcherCacheInvalidationContext {
```

Comment:
Please change this class to an interface with ProvideType annotation and add move the actual implementation to the internal package. Customers are not supposed to instantiate it or to change the state of this context.